### DISCUSSION TITLE
Azure Container Instance (ACI)

### DISCUSSION THREAD

**Discussion by jaymarvels** [advplyr/audiobookshelf#299] [General]
Is there plans for supporting this where when creating a container instance all we have to do is point to the docker image and off we go?

**Comment by advplyr** [comment]
I'm not familiar with ACI. 
The docker container runs on Windows so I'm not sure what else would be required to run on Azure. Can you provide more info?

**Reply by jaymarvels** [reply]
I'm not overly familiar with docker myself, from the ACI side of things the set up is pretty easy in terms of specifying a public docker image and it will pretty much do the rest.

Similar to this: https://hub.docker.com/r/linuxserver/booksonic-air (currently hating booksonic and wanting to move away) they provide an image:

Simply pulling lscr.io/linuxserver/booksonic-air should retrieve the correct image for your arch.

When setting up the ACI you just give it that image location and it does the rest for you which means within a couple of clicks you can have it up and running

**Reply by advplyr** [reply]
Abs also has docker images on docker hub: https://hub.docker.com/r/advplyr/audiobookshelf/tags
and can be downloaded the same way.

https://www.audiobookshelf.org/install#docker

**Reply by jaymarvels** [reply]
Ahh cool! Just tested and ACI fires it up without any issues. Next thing is to try and create it using the Azure CLI as thats the only way to make sure a volume is mounted onto the container as a share, and then hopefully within ABS I can set the library to the share, I haven't worked out a way how to pass in params when creating the container from the image to set where the storage is

**Reply by jaymarvels** [reply]
@advplyr All good! Created the container through the CLI, attached a file share hosted in Azure, went into the GUI once deployed and pointed the library to the share and it picks everything up. Already less painful than booksonic

**Reply by advplyr** [reply]
Glad to hear it!
